src/models/progress/useProgressQuery.js

```diff
@@ -0,0 +1,28 @@
+import { useRef } from 'react';
+import { useQuery } from 'react-query';
+
+import { getProgressQueryKey } from '../../constants/queryKeys';
+import { isProgressSettled } from '../../utils/progressUtils';
+import {
+  getProgress,
+  queryOptions as defaultQueryOptions,
+} from './progressQueryDefaults';
+
+export default function useProgressQuery(progressGuid, queryOptions) {
+  const isSettledRef = useRef(false);
+
+  const result = useQuery(
```

Comment:
i'm not really understanding why you're using a homemade version of useQuery here instead of useFetch. i'm probably just missing it

Reply:
I was having trouble with `useFetch` being unreliable. It was just easier to use useQuery.

Reply:
if `useFetch` is unreliable, better to fix it! what was the problem exactly? 

don't want to abandon the patterns we are introducing if we can help it